x-pack/plugins/cases/server/services/user_actions/mocks.ts

```diff
@@ -46,7 +47,7 @@ export const externalService = {
 export const originalCases = [
   { ...createCaseSavedObjectResponse(), id: '1' },
   { ...createCaseSavedObjectResponse(), id: '2' },
-];
+].map((so) => transformSavedObjectToExternalModel(so));
```

Comment:
Why that?

Reply:
Yeah thanks for pointing that out, I was getting type errors and need to look into this again 🤔

Reply:
So the part I'm confused about here is why we weren't getting type errors prior to my changes 🤔  the `createCaseSavedObjectResponse` function returns it in `SavedObject<ESCaseAttributes>` format which does not have `connector.id` (it's in the references array in the `ESCaseAttributes` format). So when we pass `originalCases` to the test here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/server/services/user_actions/index.test.ts#L715

That function `bulkCreateUpdateCase` needs a `originalCases: Array<SavedObject<CaseAttributes>>` which requires `connector.id`

Reply:
Maybe it was because the type had subcases in it?

This is what it was before my changes: `Array<SavedObject<CaseAttributes | SubCaseAttributes`

Reply:
Ok!